extension.js

```diff
@@ -36,34 +36,40 @@ exports.activate = function activate(context) {
 function createSnippet(snippetFilename, editor) {
     fs.readFile(snippetFilename, 'utf8', function(e, data) {
         console.log(data);
-        var jsonData = json.parse(data, null, false);
-        jsonData = addSelectedSnippet(jsonData, editor);
-        updateSnippetFile(snippetFilename, jsonData);
+        var jsonData = (data ? json.parse(data, null, false) : {});
+        addSelectedSnippetAsync(jsonData, editor).then(function(result) {
+            jsonData = result;
+            updateSnippetFile(snippetFilename, jsonData);
+        });
     });
 }
 
-function addSelectedSnippet(jsonData, editor) {
+function addSelectedSnippetAsync(jsonData, editor) {
     var newJsonData = {};
 
     var snippetBody = getSnippetBody(editor);
 
-    if (typeof (jsonData['//$']) !== 'undefined') {
-        // newJsonData['//$'] = jsonData['//$'];
-    }
-
-    newJsonData[DEFAULT_SNIPPET_NAME] = {
-        prefix: 'yourPrefixHere',
-        body: snippetBody,
-        description: 'Your snippet description here.'
-    };
+    return vscode.window.showInputBox({prompt:'Enter a name for your snippet'}).then(function(value) {
+        var snippetName = value;
 
-    for (var key in jsonData) {
-        if (key !== DEFAULT_SNIPPET_NAME && key !== '//$') {
-            newJsonData[key] = jsonData[key];
+        if (typeof (jsonData['//$']) !== 'undefined') {
+            // newJsonData['//$'] = jsonData['//$'];
         }
-    }
-    console.log(newJsonData);
-    return newJsonData;
+    
+        newJsonData[snippetName] = {
+            prefix: 'yourPrefixHere',
+            body: snippetBody,
+            description: 'Your snippet description here.'
+        };
```

Comment:
Yeah good idea - and then just providing the option to explicitly open a separate pane might be helpful.